interfaces/repo.go

```diff
@@ -736,9 +747,11 @@ func (r *Repository) AutoConnectCandidates(plugSnapName, plugName string) []*Slo
 	if plug == nil {
 		return nil
 	}
-	if r.ifaces[plug.Interface].AutoConnect() == false {
+
+	if (r.ifaces[plug.Interface].AutoConnect() == false) && !(isSpecialAutoConnectSnap(plug.Snap)) {
 		return nil
 	}
+
 	var candidates []*Slot
 	for _, slotsForSnap := range r.slots {
 		for _, slot := range slotsForSnap {
```

Comment:
ACK. This makes sense. I'll work on a 2 part patch here. 1 to refactor to AutoConnect(plug, slot) and the other for custom logic for my interface.